python/cudf/cudf/tests/test_multiindex.py

```diff
@@ -2079,3 +2079,20 @@ def test_multiindex_eq_other_multiindex():
     result = idx == idx
     expected = np.array([True, True])
     assert_eq(result, expected)
+
+
+@pytest.mark.parametrize(
+    "midx",
+    [
+        cudf.MultiIndex.from_product([[0, 1], [1, 0]]),
+        cudf.MultiIndex.from_tuples([(0, 1), (0, 0), (1, 1), (1, 0)]),
+        # TODO: Implement from_arrays like pandas
+        # cudf.MultiIndex.from_arrays([0, 0, 1, 1], [1, 0, 1, 0]),
+        cudf.MultiIndex(
+            levels=[[0, 1], [0, 1]], codes=[[0, 0, 1, 1], [1, 0, 1, 0]]
+        ),
+    ],
+)
```

Comment:
**issue** This parameterisation now constructs cudf objects at test collection time. Although we are not good at obeying these rules we, we prefer not to do this (see https://docs.rapids.ai/api/cudf/nightly/developer_guide/testing/#parametrization-custom-fixtures-and-pytest-mark-parametrize)

So preference is either: 
- hide these values behind a lambda `lambda: ...` which you just call inside the test 
- or (preferable) use a fixture, which is a little more setup work but can be reused:

```
@pytest.fixture(params=["product", "tuples", pytest.param("arrays", marks=pytest.mark.xfail(reason="from_arrays not yet implemented")), "levels-codes"])
def midx(request);
    if request.param == ...:
        return ...
    elif ...
```

Not also the pre-emptive marking of the arrays case as `xfail` rather than just commenting out with TODO.

Reply:
Ah I didn't know about that guidance. I converted this into a fixture instead